core/src/test/java/org/apache/iceberg/rest/TestRESTUtil.java

```diff
@@ -67,18 +70,24 @@ public void testStripTrailingSlash() {
     }
   }
 
-  @Test
-  public void testRoundTripUrlEncodeDecodeNamespace() {
+  @ParameterizedTest
+  @ValueSource(strings = {"%1F", "%2D", "%2E"})
```

Comment:
Shouldn't escaping be handled automatically?

Reply:
yes, but the test also uses the non-UTF-8 strings within the namespace, so we need to use the UTF-8 encoded string. I've added some non-UTF-8 encoded strings to the parameters list to indicate that such strings can be used as well (as long as they aren't allowed in the namespace name itself)